### PR TITLE
JSAB2-52 Troops backend

### PR DESCRIPTION
**Pull request by hongtaoTang**
troops backend was created

### CODE REVIEW

File: backend/.gitignore [hunk 1]
```diff
@@ -1,2 +1,7 @@
 node_modules
 .env
+
+# macOS
+.DS_Store
+
+.vscode/
```
Comment (evasimonyi) [inline backend/.gitignore]: as we discussed, PR acceptance criteria:
"Your modified files are related to your tasks."
this is not related.
I created a new issue on JIRA, and assigned it to you. fix gitignore, package.json and hello world typo on a new branch please, and do not modify anything else on this branch, only add your one endpoint for returning troops data.

File: backend/package.json [hunk 2]
```diff
@@ -6,7 +6,7 @@
   "scripts": {
     "test": "echo \"Error: no test specified\" && exit 1",
     "lint": "eslint .",
-    "start": "node .routers/index.js",
+    "start": "node ./routers/index.js",
```
Comment (evasimonyi) [inline backend/package.json]: same thing as I commented in the gitignore file, don't fix it in this branch

File: backend/routers/index.js [hunk 3]
```diff
@@ -6,12 +6,38 @@ const PORT = 4000;
 app.use(express.json());
 
 app.get('/', (req, res) => {
-  res.status(200).send('Hellow world');
+  res.status(200).send('Hello world');
```
Comment (evasimonyi) [inline backend/routers/index.js]: fixing this typo is not related to your task

File: backend/routers/index.js [hunk 4]
```diff
@@ -6,12 +6,38 @@ const PORT = 4000;
...
 });
 
 app.use((err, req, res, next) => {
   res.status(500).send('Something broke!');
   next(err);
 });
 
+const dummyData = {
```
Comment (evasimonyi) [inline backend/routers/index.js]: please don't call it like this, it's obvious that it's just mocked data. rename it to myTroops, or something similar

File: backend/routers/index.js [hunk 5]
```diff
@@ -6,12 +6,38 @@ const PORT = 4000;
...
+      attack: 1,
+      defence: 1,
+      started_at: 12345789,
+      finished_at: 12399999,
+    },
+  ],
+};
+
+app.get('/kingdom/troops', (req, res) => {
+  res.status(500).send(dummyData);
```
Comment (evasimonyi) [inline backend/routers/index.js]: you send back the data with the 500 status code? it should be 200, isn't it?

**Comment by evasimonyi** [comment]
@GuJialu @Kyya @HanqingZ you all approved this line:

`res.status(500).send(dummyData);`

If everything is fine, and we are sending back the requested data, why should we set the status code to 500 instead of 200? 

Also, multiple not task-related files were modified, and none of you made a comment about it.

Next time please pay much more attention to the reviews and request changes!!

File: backend/routers/index.js [hunk 6]
```diff
@@ -9,6 +9,32 @@ app.get('/', (req, res) => {
...
+      defence: 1,
+      started_at: 12345789,
+      finished_at: 12399999,
+    },
+  ],
+};
+
+app.get('/kingdom/troops', (req, res) => {
+  res.status(200).send(myTroops);
+});
```
Comment (Kyya) [inline backend/routers/index.js]: May be we can write these handlers into separate files like this:
```js
const { Router } = require('express');
const router = Router();

function getTroops(req, res, next) {
  return res.send(myTroops);
}

// other handlers can be here

router.get('/', getTroops);

module.exports = router;
```

And then import these in app.js
```js
import { troops } from './routes';
// ... other middlewares
app.use('/troops', troops);
```

Comment (evasimonyi) [reply backend/routers/index.js]: yes please, great suggestion!

Comment (evasimonyi) [reply backend/routers/index.js]: instead of approving and just commenting, you can also ask for change during your review, do this more time! :)

File: backend/index.js [hunk 7]
```diff
@@ -0,0 +1,20 @@
+const express = require('express');
+
+const app = express();
+const PORT = 4000;
+const { troops } = require('./routers/index');
```
Comment (Kyya) [inline backend/index.js]: It will import the `index.js` default when you want to import from a folder.
```js
const { troops } = require('./routers');
```
